public/javascript/components/surveyController.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Survey Controller
+ */
+
+import $ from 'jquery';
+import firebase from "firebase/app";
+import "firebase/database";
+
+// Submit button on click
+$('.survey__submit').click(function(e) {
+  e.preventDefault();
+
+  checkAllButtonsClicked()
+})
+
+// Validation to see if buttons are all checked
+const checkAllButtonsClicked = () => {
+  let checkState = true
+
+  $('input:radio').each(function() { 
+    let name = $(this).attr('name');
+
+    if($("input:radio[name=" + name + "]:checked").length == 0) {
```

Comment:
Since we're looping through each radio button, this would execute twice per question, I'm thinking about looping through each question instead and then we can use the `this` context to get it's radio buttons like

```javascript
$('survery__question').each(function() {
  if !$(this).find('input:radio').is(':checked') {
    checkState = false;
  }
// OR
 !$(this).find('input:radio').is(':checked') && checkState = false
}
```

What do you think about that? 💭

Reply:
👍🏾